File: C#/CryptoLib/Vigenere.cs
using System.Collections.Generic;
using System.Linq;

namespace CryptoLib
{
    public static class Vigenere
    {
        private static int Mod(int a, int b)
        {
            return (a % b + b) % b;
        }

        private static string Cipher(string input, string key, bool encipher)
        {
            if (string.IsNullOrEmpty(key))
            {
                return "Error: You must enter a keyword";
            }
            for (int i = 0; i < key.Length; ++i)
                if (!char.IsLetter(key[i]))
                {
                    return "Error: Non-letter character entered in keyword";
                }

            string output = string.Empty;
            int nonAlphaCharCount = 0;

            for (int i = 0; i < input.Length; ++i)
            {
                if (char.IsLetter(input[i]))
                {
                    bool cIsUpper = char.IsUpper(input[i]);
                    char offset = cIsUpper ? 'A' : 'a';
                    int keyIndex = (i - nonAlphaCharCount) % key.Length;
                    int k = (cIsUpper ? char.ToUpper(key[keyIndex]) : char.ToLower(key[keyIndex])) - offset;
                    k = encipher ? k : -k;
                    char ch = (char)((Mod(((input[i] + k) - offset), 26)) + offset);
                    output += ch;
                }
                else
                {
                    output += input[i];
                    ++nonAlphaCharCount;
                }
            }

            return output;
        }

        public static string Encrypt(string input, string key)
        {
            return Cipher(input, key, true);
        }

        public static string Decrypt(string input, string key)
        {
            return Cipher(input, key, false);
        }
        
    }
}

File: Java/Cryptography/src/com/sykes368/lib/Vigenere.java
package com.sykes368.lib;

/** Vigenere Cipher
 * @author Sykes368
 * @since 1.0
 */
public class Vigenere {
    public static String encrypt(String keyword, String plaintext) {
       return vigenere(plaintext, keyword, false);
    }

    public static String decrypt(String keyword, String ciphertext) {
       return vigenere(ciphertext, keyword, true);
    }

    private static String vigenere(String input, String keyword, boolean decode) {
        keyword = keyword.toUpperCase().replace(" ", "");
        if (keyword == null) {
            return "Error: You must enter a keyword";
        }
        for (int i = 0; i < keyword.length(); ++i) {
            if (!Character.isLetter(keyword.charAt(i))) {
                return "Error: Non-letter character entered in keyword";
            }
        }
        if (input == null || !input.matches("^[ A-Za-z]+$")) {
            return "ERROR: Input cannot be empty and Only letters and spaces are allowed as input";
        }

        input = input.toUpperCase().replace(" ", "");
        String key = generateKey(input, keyword);
        StringBuilder output = new StringBuilder();

        for (int i = 0 ; i < input.length(); i++)
        {
            int x;
            if (decode) {
                x = (input.charAt(i) - key.charAt(i) + 26) %26;
            } else {
                x = (input.charAt(i) + key.charAt(i)) %26;
            }

            x += 'A';
            output.append((char) (x));
        }
        return output.toString();
    }

    private static String generateKey(String input, String key)
    {
        int x = input.length();

        StringBuilder keyBuilder = new StringBuilder(key);
        for (int i = 0; ; i++) {
            if (x == i) {
                i = 0;
            }
            if (keyBuilder.length() == input.length()) {
                break;
            }
            keyBuilder.append(keyBuilder.charAt(i));
        }
        key = keyBuilder.toString();
        return key;
    }
}
